src/cider/nrepl/middleware/debug.clj

```diff
@@ -0,0 +1,67 @@
+(ns cider.nrepl.middleware.debug
+  (:require [clojure.tools.nrepl.transport :as transport]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.misc :refer [response-for]]
+            [debugger.core :refer [break]]))
+
+;;;; ## Interaction with the client
+;;; `wrap-debug` receives an initial message from the client, stores
+;;; it in `debugger-message`, and `breakpoint` answers it.
+(def debugger-message nil)
+
+(defn wrap-debug [h]
+  (fn [{:keys [op force] :as msg}]
+    (if (= "init-debugger" (:op msg))
+      (if (and debugger-message (not force))
+        (transport/send (:transport msg)
+                        (response-for msg :status :done))
+        (do (when debugger-message
```

Comment:
Actually, it stores the first request it receives, and immediately closes all subsequent ones. But it makes no difference in the end. We could store the most recent instead, and close the older one, and the final effect would be the same.

Regardless. I think I'll rewrite `cider-debug-defun-at-point` the way we discussed above, so that this whole thing will no longer be necessary.